r/R/compute.R

```diff
@@ -127,15 +127,15 @@ match_arrow.Array <- function(x, table, ...) {
   if (!inherits(table, c("Array", "ChunkedArray"))) {
     table <- Array$create(table)
   }
-  Array$create(call_function("index_in_meta_binary", x, table))
+  call_function("index_in_meta_binary", x, table)
 }
 
 #' @export
 match_arrow.ChunkedArray <- function(x, table, ...) {
   if (!inherits(table, c("Array", "ChunkedArray"))) {
     table <- Array$create(table)
   }
-  shared_ptr(ChunkedArray, call_function("index_in_meta_binary", x, table))
+  call_function("index_in_meta_binary", x, table)
 }
```

Comment:
This can be `match_arrow.ChunkedArray <- match_arrow.Array` now since they're the same. There are probably a few others like this (I see `Ops` also, for example).

Reply:
We can go further when tackling this: https://issues.apache.org/jira/browse/ARROW-10089